### PR TITLE
Raft receive append entries tweaks

### PR DESCRIPTION
**Pull request by jumaffre**
Small tweaks to the Raft's receive append entries logic after trying to implement https://github.com/microsoft/CCF/issues/1303 

- Check that `r.prev_idx` is not in the future
- Move leader assignment earlier, when we restart the election timeout
- Do not throw if deserialision fails
- Comments

### CODE REVIEW

File: src/consensus/raft/raft.h [hunk 1]
```diff
@@ -635,8 +637,17 @@ namespace raft
         return;
       }
 
+      // If the terms match up, it is sufficient to convince us that the sender
+      // (leader) is legit in our term
```
Comment (achamayou) [inline src/consensus/raft/raft.h:641]:
```suggestion
      // (leader) is leader in our term
```

**Review by achamayou — APPROVED** [review]

File: src/consensus/raft/raft.h [hunk 2]
```diff
@@ -624,7 +626,7 @@ namespace raft
         {
           LOG_DEBUG_FMT(
             "Recv append entries to {} from {} but our log at {} has the wrong "
-            "term (ours: {}, theirs: {})",
+            "previous term (ours: {}, theirs: {})",
```
Comment (ashamis) [inline src/consensus/raft/raft.h:629]: nit: view

File: src/consensus/raft/raft.h [hunk 3]
```diff
@@ -635,8 +637,17 @@ namespace raft
         return;
       }
 
+      // If the terms match up, it is sufficient to convince us that the sender
```
Comment (ashamis) [inline src/consensus/raft/raft.h:640]: term => view

File: src/consensus/raft/raft.h (hunk 2 above)
Comment (jumaffre) [reply src/consensus/raft/raft.h:629]: `raft.h` still uses the Raft-specific term-inology (no pun intended) for both comments and logs. I think we should address this separately if we want to? The `raft_consensus.h` adapter provides the translation between both.

**Pull request merged**